src/packages/DAO/elections.dao.ts

```diff
@@ -1,13 +1,23 @@
 "use server";
 
 import prisma from "@lib/prisma";
-import { createElectionValidator } from "@packages/dto/elections.dto";
+import {
+  createElectionValidator,
+  electionSchema,
+  updateElectionSchema,
+} from "@packages/dto/elections.dto";
 import type { CreateElection } from "@packages/entities/elections";
+import { BadRequestError, NotFound } from "@packages/utils/error";
 import { ElectionStatus } from "@prisma/client";
 import { revalidatePath } from "next/cache";
 import { redirect } from "next/navigation";
 import { z } from "zod";
 
+interface FindElection {
+  name: string;
+  status: string;
+}
```

Comment:
You need to infer this type from the schema, no need to create it

```suggestion
type FindElection = z.infer<typeof createElectionValidator>;
```

To read:
- https://zod.dev/?id=type-inference
- https://stackoverflow.com/questions/73043223/infer-type-from-key-of-object-inside-an-array-zod

Reply:
UP!

Reply:
change ehre [569ba6f](https://github.com/podcodar/voting-system/pull/67/commits/569ba6fbee9158752a002ace28c11ec95badc1c9)

Reply:
This approach avoids having to redo something unnecessary. I really enjoyed learning and understanding this; it's something I was always unsure about. @frattezi @marco-souza 💯